apollo-federation/src/error/mod.rs

```diff
@@ -615,6 +621,39 @@ impl SingleFederationError {
             },
         }
     }
+
+    pub(crate) fn unwrap_subgraph_error(self, expected_subgraph: &str) -> Self {
```

Comment:
Can we leave a doc string here to explain that this method is to avoid having the type recur more than once? And that, ideally, the composition error in the future doesn't need to recur at all?

Reply:
Makes sense, added the TODO/doc strings in https://github.com/apollographql/router/pull/7436/commits/7a6e4b4fa192aef65895bb1c8bdc6c6c3f10def8